### DISCUSSION TITLE
scv.pl.scatter function

### DISCUSSION THREAD

**Discussion by huizizhang949** [theislab/scvelo#883] [Q&A] [answered]
Hello,

I want to plot the (filtered and normalized) phase portrait for some genes in particular using the raw plt.plot function. For example,

> plt.plot(adata.layers['spliced'].todense()[:,adata.var_names=='Igfbpl1'],
        adata.layers['unspliced'].todense()[:,adata.var_names=='Igfbpl1'],'o')

<img width="572" alt="image" src="https://user-images.githubusercontent.com/79323958/170591034-6d395911-065e-47ca-8475-7dbb8318d596.png">

But the phase portrait is different from the one produced by 

> scv.pl.scatter(adata, basis='Igfbpl1', frameon=True)

<img width="632" alt="image" src="https://user-images.githubusercontent.com/79323958/170591063-4b8ddec2-b7ee-45b0-85d0-f715f5b89bbb.png">

The one shown by scv.pl.scatter seems to be more close to the desired 'almond' shape and x limits and y limits are also different. Is there any step in scv.pl.scatter that transforms or normalizes the values in adata.layers['spliced']?

Thank you!

**Comment by WeilerP** [comment, marked as answer]
@Huizi949-z, `scvelo.plotting.scatter` plots the smoothed impression by default, _i.e._, `Mu` and `Ms`. Otherwise, the data is too noisy to robustly infer rates.

**Reply by huizizhang949** [reply]
Thanks for the reply!! 

Could you please tell me how `Mu` and `Ms` are computed? Where can I extract these values from adata? I look into the source code for the function here https://github.com/theislab/scvelo/blob/1805ab4a72d3f34496f0ef246500a159f619d3a2/scvelo/plotting/scatter.py#L20

but I could not find where and how `Mu` and `Ms` get involved.

Thank you!

**Reply by WeilerP** [reply]
@Huizi949-z, please have a look at the paper, tutorials, and docs. Everything should be explained there.
First order moments (`Mu` and `Ms`) are calculated by smoothing a cell's gene expression over its neighbors. This is done when calling `scvelo.preprocessing.moments()`. Consequently, you'll find the corresponding entries in `adata.layers`.
